lib/src/util/vulnerabilitySeverityRatings.ts

```diff
@@ -67,10 +69,9 @@ export const getVulnerabilitySeverityRating = (cvssScore: VulnerabilityScore | n
  * (Internal use, not exported to library users)
  * Maps vulnerability CVSS severity ratings to their corresponding score ranges string representation.
  */
-export const vulnerabilitySeverityScoreRangeMap = Object.freeze({
-  critical: '9.0—10.0',
-  high: '7.0—8.9',
-  medium: '4.0—6.9',
-  low: '0.1—3.9',
-  none: '0.0'
-});
+export const vulnerabilitySeverityScoreRangeMap = Object.freeze(
+    vulnerabilitySeverityAndScoreRangeList.reduce((acc, range) => {
+      acc[range.severity] = `${range.min}—${range.max}`;
+      return acc;
+    }, {} as Record<VulnerabilitySeverityRating, string>)
```

Comment:
Mutating the input within a `reduce` operation is a bit of a violation of its intended use.  If you want to use a mutable style to implement this, I'd rather you just use a `for` loop than a `reduce` call.

Reply:
> Mutating the input within a reduce operation is a bit of a violation of its intended use. If you want to use a mutable style to implement this, I'd rather you just use a for loop than a reduce call.

But my top preference would probably be to use ramda.  Probably map to tuples and then use `fromPairs`

Reply:
> > Mutating the input within a reduce operation is a bit of a violation of its intended use. If you want to use a mutable style to implement this, I'd rather you just use a for loop than a reduce call.
> 
> But my top preference would probably be to use ramda. Probably map to tuples and then use `fromPairs`

Fixed that here: [9642cc5](https://github.com/sonatype/sonatype-react-shared-components/pull/1202/commits/9642cc5fe1ab3e31c3f982313d993970c398730c)